src/toolkits/object_detection/object_detector.hpp

```diff
@@ -219,6 +219,8 @@ class EXPORT object_detector: public ml_model_base {
                                            const std::string& output_type);
   static gl_sframe convert_types_to_sframe(const variant_type& data,
                                            const std::string& column_name);
+  static gl_sarray canonicalize_annotations(gl_sarray& data);
```

Comment:
Let's pass by value. It could be unexpected that this function can mutate its argument. (https://google.github.io/styleguide/cppguide.html#Reference_Arguments)